### PR TITLE
update codecov shield

### PR DESCRIPTION
**Pull request by sfc-gh-mkeller**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Just updating the codecov shield to show actual coverage

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Snowflake SQLAlchemy
 ********************************************************************************
 
-.. image:: https://codecov.io/gh/snowflakedb/snowflake-sqlalchemy/branch/master/graph/badge.svg
+.. image:: https://codecov.io/gh/snowflakedb/snowflake-sqlalchemy/branch/main/graph/badge.svg?token=1kmCWB3WRS
```
Comment (sfc-gh-aling) [inline README.rst:4]: should this token be included? just want to make sure no credential is leaked

Comment (sfc-gh-mkeller) [reply README.rst:4]: I hope it's not sensitive, the whole `.rst` copy-paste is generated by codecov

Comment (sfc-gh-mkeller) [inline README.rst:4]: Let's try without it
```suggestion
.. image:: https://codecov.io/gh/snowflakedb/snowflake-sqlalchemy/branch/main/graph/badge.svg
```

**Comment by sfc-gh-mkeller** [comment]
@sfc-gh-aling it seems like the token is not necessary for the shield to work

**Review by sfc-gh-aling — APPROVED** [review]
LGTM

**Pull request merged**